### PR TITLE
ISO 3166-1 for US subdivisions

### PR DESCRIPTION
**Pull request by mborsetti**
Per [here](https://en.wikipedia.org/wiki/ISO_3166-2:US#Subdivisions_included_in_ISO_3166-1), besides being included as subdivisions of the United States in ISO 3166-2, the following US outlying areas are also officially assigned their own country codes in ISO 3166-1:
* American Samoa
* Guam 
* Northern Mariana Islands
* Puerto Rico
* United States Minor Outlying Islands
* Virgin Islands, U.S.

This PR implements ISO 3166-1 country entries (as aliases) for these areas.

Please note that pre-commit would not work until I remarked the `"multi_line_output = 9"` line of `"[tool.isort]"` in `pyproject.toml`. Without this, black and isort would fight over how to format `countries\__init__.py` and pre-commit would fail.  I am not sure how it's best to resolve this, but I saw in isort's documentation that `"multi_line_output"` defaults to 3 when used in conjunction with black.

### CODE REVIEW

**Comment by mborsetti** [comment]
Well, the edit to `pyproject.toml` worked differently in my environment than on CI!.

Unfortunately I found no way around the issue of black and isort fighting with each other than to exclude `holidays/countries/__init__.py` from isort. 

Will add a TODO comment and leave it as it is for now.

**Comment by mborsetti** [comment]
Not sure what happened, but in addition to the IDE giving different results than CI with isort, I lost a bunch of edits in README.rst. I am sure it's use error, and hopefully this is the last push for this PR.

Incidentally, I found IRL that finding a class called "UnitedStates" or "Eswatini" or similar in the Debugger is very confusing -- Eswatini what? What is the Eswatini object supposed to be?  As such, for these new aliases I chose to call the base class HolidaysAS, HolidaysVI, to make it very clear the function of an object ; IMHO we should consider backporting this for an improved developer experience.

File: holidays/countries/united_states.py [hunk 1]
```diff
@@ -424,6 +436,12 @@ def _populate(self, year):
             elif year >= 1937:
                 self[date(year, OCT, 12)] = name
 
+        # Commonwealth Cultural Day in Northern Mariana Islands
+        if self.subdiv == "MP":
+            self._add_with_observed(
+                date(year, MAR, 24), "Commonwealth Cultural Day"
+            )
+
```
Comment (KJhellico) [inline holidays/countries/united_states.py:444]: Duplicate with lines 242-247?

File: holidays/countries/united_states.py [hunk 2]
```diff
@@ -424,6 +436,12 @@ def _populate(self, year):
             elif year >= 1937:
                 self[date(year, OCT, 12)] = name
 
+        # Commonwealth Cultural Day in Northern Mariana Islands
+        if self.subdiv == "MP":
+            self._add_with_observed(
+                date(year, MAR, 24), "Commonwealth Cultural Day"
```
Comment (KJhellico) [inline holidays/countries/united_states.py:442]:
```suggestion
                date(year, OCT, 10), "Commonwealth Cultural Day"
```

File: holidays/countries/united_states.py [hunk 3]
```diff
@@ -424,6 +436,12 @@ def _populate(self, year):
...
+            )
```
Comment (KJhellico) [inline holidays/countries/united_states.py:443]:
```suggestion
            self[
                date(year, OCT, 1) + rd(weekday=MO(+2)
            ] = "Commonwealth Cultural Day"
```

**Comment by dr-prodigy** [comment]
Great job guys @mborsetti @KJhellico ! Thanks as usual, I merged this :+1:

**Pull request merged**